modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1114,6 +1115,8 @@ public void registerSqlTypeNameToClassMappings(Map<String, Class<?>> mappings) {
 
     @Override
     public void registerSqlTypeToSqlTypeNameOverrides(Map<Integer, String> overrides) {
+        //Step 1 Geos-7533 put the types in dialect
+        overrides.put(4, "int4");
```

Comment:
Please use ``Types.INTEGER`` instead of ``4``.

Reply:
hard coding removed, this is addressed